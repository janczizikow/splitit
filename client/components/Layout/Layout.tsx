import React from "react";
import styled from "@emotion/styled";
import Meta from "../Meta";
import Header from "../Header";
import Footer from "../Footer";

interface LayoutProps {
  children: React.ReactNode;
}

interface SiteContentProps {
  fullScreen?: boolean;
}

type Props = LayoutProps & SiteContentProps;

const Site = styled.div`
  display: flex;
  flex-direction: column;
  min-height: 100vh;
`;

const SiteContent = styled.main`
  flex: 1;
  display: ${(p: SiteContentProps) => p.fullScreen && "flex"};
  flex-direction: ${(p: SiteContentProps) => p.fullScreen && "column"};
`;

const Layout: React.FunctionComponent<Props> = ({ fullScreen, children }) => (
  <Site>
    <Meta />
    <Header />
    <SiteContent fullScreen={fullScreen}>{children}</SiteContent>
    <Footer />
  </Site>
);

export default Layout;
