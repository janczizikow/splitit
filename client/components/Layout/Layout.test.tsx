import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Layout from "./Layout";

describe("<Layout />", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<Layout children={<span>Child</span>} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
