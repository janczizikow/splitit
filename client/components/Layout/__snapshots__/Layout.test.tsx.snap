// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> matches snapshot 1`] = `
<Styled(div)>
  <Meta />
  <Header />
  <Styled(main)>
    <span>
      Child
    </span>
  </Styled(main)>
  <Footer />
</Styled(div)>
`;
